app/src/main/java/org/jellyfin/mobile/ui/screens/connect/ServerSelection.kt

```diff
@@ -53,31 +53,60 @@ import androidx.compose.ui.text.input.KeyboardType
 import androidx.compose.ui.unit.dp
 import kotlinx.coroutines.launch
 import org.jellyfin.mobile.R
+import org.jellyfin.mobile.app.ApiClientController
 import org.jellyfin.mobile.setup.ConnectionHelper
 import org.jellyfin.mobile.ui.state.CheckUrlState
 import org.jellyfin.mobile.ui.state.ServerSelectionType
 import org.jellyfin.mobile.ui.utils.CenterRow
-import org.jellyfin.sdk.model.api.ServerDiscoveryInfo
-import org.koin.androidx.compose.get
+import org.koin.compose.koinInject
 
 @OptIn(ExperimentalComposeUiApi::class)
+@Suppress("LongMethod")
 @Composable
 fun ServerSelection(
     showExternalConnectionError: Boolean,
-    connectionHelper: ConnectionHelper = get(),
+    apiClientController: ApiClientController = koinInject(),
+    connectionHelper: ConnectionHelper = koinInject(),
     onConnected: suspend (String) -> Unit,
 ) {
     val coroutineScope = rememberCoroutineScope()
     val keyboardController = LocalSoftwareKeyboardController.current
     var serverSelectionType by remember { mutableStateOf(ServerSelectionType.ADDRESS) }
     var hostname by remember { mutableStateOf("") }
+    val serverSuggestions = remember { mutableStateListOf<ServerSuggestion>() }
     var checkUrlState by remember<MutableState<CheckUrlState>> { mutableStateOf(CheckUrlState.Unchecked) }
     var externalError by remember { mutableStateOf(showExternalConnectionError) }
 
-    val discoveredServers = remember { mutableStateListOf<ServerDiscoveryInfo>() }
+    // Prefill currently selected server if available
+    LaunchedEffect(Unit) {
+        val server = apiClientController.loadSavedServer()
+        if (server != null) {
+            hostname = server.hostname
+        }
+    }
+
+    // Suggest saved servers
+    LaunchedEffect(Unit) {
+        apiClientController.loadPreviouslyUsedServers().mapTo(serverSuggestions) { server ->
+            ServerSuggestion(
+                name = server.hostname,
+                address = server.hostname,
+                lastUsedTimestamp = server.lastUsedTimestamp,
+            )
+        }
+    }
+
+    // Server discovery
     LaunchedEffect(Unit) {
         connectionHelper.discoverServersAsFlow().collect { serverInfo ->
-            discoveredServers.add(serverInfo)
+            serverSuggestions.add(
+                index = 0,
+                ServerSuggestion(
+                    name = serverInfo.name,
+                    address = serverInfo.address,
+                    lastUsedTimestamp = System.currentTimeMillis(),
+                ),
+            )
```

Comment:
We should check if the suggestion is already the current server or a previously used server to avoid duplicates. I'd also argue we should not set the lastUsedTimestamp to now but to null/0 instead.

Reply:
But that implies that every time you click an entry you'd have to remove it from the suggestions and add back the currently filled server to the suggestions if applicable. That's a lot of complexity for something we could just ignore.
Having the server in both the selection and suggestions panel isn't a huge issue anyway. Duplicates within the suggestions should be eliminated, though, agreed.

I set the lastUsedTimestamp to the current time on purpose, since that's what the servers should be sorted by. It's true that it's not necessarily a "usage time" then, but 0/null only makes things worse.

Reply:
> But that implies that every time you click an entry you'd have to remove it from the suggestions and add back the currently filled server to the suggestions if applicable. That's a lot of complexity for something we could just ignore.
Having the server in both the selection and suggestions panel isn't a huge issue anyway. Duplicates within the suggestions should be eliminated, though, agreed.

Fair enough

> I set the lastUsedTimestamp to the current time on purpose, since that's what the servers should be sorted by. It's true that it's not necessarily a "usage time" then, but 0/null only makes things worse.

The rename to just "timestamp" works for that 👍